### ISSUE TITLE
Multiple `output-format` arguments with a file output produces all files, but only the last argument's file is filled.

### ISSUE THREAD

**Issue by rhtenhove** [PyCQA/pylint#8147]
### Bug description

All files are created specified in `--output-format`s, but only the last file has content. This happens with other output formats as well, such as JUnit. This may cause certain systems (CI for example) and people to see perfect code (no news is good news, right?) while problems are shown in a different format.

### Command used

```shell
pylint --output-format=text:./pylint.txt --output-format=text:./pylint2.txt <package>

# You can add more, i.e. --output-format=text:./pylint3.txt
```

### Pylint output

```shell
Multiple files, last one with content
```

### Expected behavior

All files with their expected content

_or_

One file, with content

_or_

An error that the argument was supplied multiple times

### Pylint version

```shell
pylint 2.15.9
astroid 2.12.13
Python 3.11.0 (main, Oct 24 2022, 18:26:48) [MSC v.1933 64 bit (AMD64)]
```
